lib/src/base/mixins/network.dart

```diff
@@ -89,18 +118,47 @@ mixin NetworkBaseMixin<T, S extends NetworkStateBase<T>> on BlocBase<S> {
     }
   }
 
+  /// [onLoadAsync] is called  internally  when [load] method is called.
+  ///
+  /// Must be overridden when [NetworkBloc] or [NetworkCubit] is created. When extending [NetworkBloc] or [NetworkCubit], the IDE will warn that this method requires an override due to the missing implementation.
+  ///
   Future<T> onLoadAsync();
 
+  ///  Can optionally be overridden when creating [NetworkBloc] or [NetworkCubit].
+  ///
+  /// [onUpdateAsync] is called internally when [updateAsync] is called.
+  /// Example usage:
+  /// ```dart
+  ///   @override
+  ///   Future<Data> onUpdateAsync(Data updatedData) async {
+  ///     return someRepository.update(updatedData);
+  ///   }
+  /// ```
+  ///
   Future<T> onUpdateAsync(T updatedData) => Future.value(updatedData);
 
+  /// Returns the state with modified `status`.
+  ///
+  /// It accepts [state] and [reason] that has
+  /// been updated (typically via `copyWithSuccess`) and ensures that the `status`
+  /// field is set to `NetworkStatus.success`.
+  ///
   S onStateChanged(DataChangeReason reason, S state) =>
       state.copyWith(status: NetworkStatus.success) as S;
 
-  //additional methods
+  // Additional methods
+  ///
+  /// Returns the first `state` if the [NetworkStatus] is **not** `loading`.
+  ///
+  ///  Listens to the stream and stops after the first match.
```

Comment:
Rewrite to  more strict description  
Example:
Start listening state stream and return first stream where the [NetworkStatus] is **not** `loading`.
"Attention" make sure you provoke event to avoid endless listening